src/main/java/seedu/address/model/person/Room.java

```diff
@@ -21,15 +22,33 @@
      */
     public Room(String room) {
         requireNonNull(room);
-        checkArgument(isValidRoom(room), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidRoom(room), getMessageConstraints());
         value = room;
     }
 
     /**
      * Returns true if a given string is a valid Room number.
      */
     public static boolean isValidRoom(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        int room = Integer.parseInt(test);
+        boolean isRoomValid = room % 100 <= roomPref % 100 && room % 100 > 0;
+        boolean isFloorValid = Math.floorDiv(room, 100) <= Math.floorDiv(roomPref, 100)
+                && Math.floorDiv(room, 100) > 0;
+        return isFloorValid && isRoomValid;
+    }
+
+    public static String getMessageConstraints() {
+        int floorMax = Math.floorDiv(roomPref, 100);
+        int roomMax = roomPref % 100;
+        return MESSAGE_CONSTRAINTS + "\nFloor : 1 to " + String.valueOf(floorMax)
+                + "\nRoom No : 1 to " + String.valueOf(roomMax);
+    }
```

Comment:
Perhaps can declare `MIN, MAX_FLOOR_NUMBER` and `MIN, MAX_ROOM_NUMBER` as a static variable and include it in the `MESSAGE_CONSTRAINTS`?